docs/modules/ROOT/pages/documentation/microprofile/config/cloud/azure.adoc

```diff
@@ -0,0 +1,116 @@
+
+NOTE: If you have access to multiple tenants and what to select a specific tenant to register your application, use the Directory + subscription filter in the top menu.
+
+image:microprofile/config/cloud/azure/application-registration.png[Application Registration]
+
+Then provide a name for your application.  Select the appropriate account types and select Register to complete your application registration. Once the application registration is completed, the service principal is automatically created in your home tenant. 
+
+image:microprofile/config/cloud/azure/application-registration-configuration.png[Application Registration Configuration]
+
+## Application API Permissions
+You will need to allow your application full access to the Azure Key Vault service to retrieve the secrets stored within it. To do this you will need to configure your application's API permissions. You can learn more about permissions and consent https://docs.microsoft.com/en-gb/azure/active-directory/develop/v2-permissions-and-consent[here]. 
+
+Once you have registered your application and created a service principle. Select your application, under Manage section select API permissions, select Add permission, select Azure Key Vault for list of Microsoft APIs, then under types of permission select Delegated permissions, under Permission select _user_impersonation_ and finally select Add permissions.   
+
+image:microprofile/config/cloud/azure/API-permissions.png[API Permissions]
+
+## Add Credentials
+
+Credentials allow your application to authenticate as itself. Currently Azure supports two options as credentials: client secrets (a string) and certificates (public key). Currently, Payara only supports certificate as a credential as it provides a higher level of assurance than a client secret.
+
+To upload for own certificate for authentication. Select your application, under Manage section select Certificates & secrets, select Upload certificate, select the certificate you would like to upload and select Add. Once the certificate is successfully uploaded it should generate a Thumbprint, this will be required when we configure the Azure Config Source.
+
+image:microprofile/config/cloud/azure/certificate-upload.png[Certificate Upload]
+
+## Create a Key Vault
+
+To securely store your secrets you will need to create a key vault. To create a key valut using the Azure portal, search for key vaults and select it, select Add and provide relvant information on the Create key vault page and finally select Review + create.  
+
+NOTE: Although key vault allows storing of keys, secrets, and certificates, Azure Config Source currently only support secrets.
+
+image:microprofile/config/cloud/azure/key-vault-creation.png[Key Vault Creation]
+
+You can see from the above diagram I have only configured the Basic section and other section such as Access policy and Networking has been left to their defaults. You can confguire these section yourslef to fit your requirments.
```

Comment:
The documentation  is an official manual, so it should not be written as a personal piece of writing, instead of saying, "You can see from the above diagram, I  have".. it is better to state" In the above diagram the Basic section has been configured and the remaining sections have been left to their default settings". Also, minor typo at "configure"